solutions/cf-1216C.mdx

```diff
@@ -0,0 +1,232 @@
+---
+id: cf-1216C
+source: CF 1216C
+title: D3C - White Sheet
+author: David Zhang
+---
+
+## Info/Tips
+Especially with rectangle geometry, make a good habit of drawing patterns and observations on a scratch sheet of paper or whiteboard. It helps a lot - David Zhang.
+Also if you don't understand the explanation, draw each possibility (Top, Bottom, Left and Right) on a sheet of paper and try to see the connections.
+
+[Official Analysis (C++)](https://codeforces.com/blog/entry/69954)
+
+## Explanation
+
+There are many ways to solve the problem, but the easiest way is to imagine cutting the white sheet. 
+
+Everytime the white sheet is covered by a black sheet, we can imagine cutting the intersection between the black and white sheets. Then, if the area of the final white sheet after being cut by the two rectangles is greater than zero, there is a portion of the white sheet that is visible.
+
+However, when considering cutting the sheet, the portion that is being cut must cut from entirely x1->x2 or y1->y2 or else there will be a portion of the white sheet still shown. We only cut when a black sheet completely covers the entire width or length of the white sheet, otherwise there will still be bits and pieces shown.
+
+
+There are 4 instances of which a white sheet can cover a black sheet.
+
+$\textbf{Top: }$ From the top, if the white sheet is completely covered from x1 to x2 and intersects some part of a black sheet, then change the y2 (top of the rectangle) to the bottom of the black rectangle.
+
+$\textbf{Bottom: }$ From the bottom, if the white sheet is completely covered from x1 to x2 and intersects some part of a black sheet, then change the y1 (bottom of the rectangle) to the top of the black rectangle.
+
+$\textbf{Left: }$ From the left, if the white sheet is completely covered from y1 to y2 and intersects some part of a black sheet, then change the x1 (left edge of the rectangle) to the right edge of the black rectangle.
+
+$\textbf{Right: }$ From the right, if the white sheet is completely covered from y1 to y2 and intersects some part of a black sheet, then change the x2 (right edge of the rectangle) to the left edge of the black rectangle.
+
+Also, keep in mind to make sure that x1 is always < x2 and the same with y1 and y2.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(1)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+struct Rect {
+	int x1, y1, x2, y2;
+	int area() {
+		return (x2 - x1) * (y2 - y1);
+	}
+};
+
+/*
+ * Main Idea: If B intersects entirely in the x or y direction, cut it.
+ * This method cuts rectangle A based on rectangle B. (A - white sheet, B - black sheet)
+ * We can cut rectangle A if B covers all of x1->x2 or y1->y2.
+ */
+Rect cut(Rect a, Rect b) {
+	// If B cuts A from the left side
+	if (a.x1 >= b.x1 && b.x2 >= a.x1 && b.y1 <= a.y1 && b.y2 >= a.y2) {
+		a.x1 = b.x2;
+		a.x2 = max(a.x2, b.x2);  // If b covers A entirely
```

Comment:
capitalize the `b`'s to be consistent

do the same with the others

Reply:
you mean in the comment right?

Reply:
yep